src/test/test_gradientcategorial_ruleset.js

```diff
@@ -0,0 +1,68 @@
+var OncoprintRuleSet = require("../js/oncoprintruleset.js");
+var assert = require("chai").assert;
+var $ = require('jquery');
+
+describe("GradientCategoricalRuleSet", function() {
+
+    var mixParams = {
+        type: 'gradient+categorical',
+        legend_label: "this is a label",
+        value_key: "profile_data",
+        value_range: [1,8],
+        value_stop_points: [1,2,8],
+        colors: [[255,0,0],[0,0,0],[0,255,0]],
+        null_color: 'rgba(224,224,224,1)',
+        category_key: "category"
+    }
+
+    var categoryDatum1 = {
+        category: ">8",
+        profile_data: 8,
+        truncation: ">"
+    }
+    
+    var gradientDatumLargest = {
+        category: undefined,
+        profile_data: 8,
+        truncation: ""
+    }
+
+    var gradientDatumSmallest = {
+        category: undefined,
+        profile_data: 1,
+        truncation: undefined
+    }
+
+    var naDatum = {
+        category: undefined,
+        profile_data: undefined,
+        truncation: undefined
+    }
+
+    it("Is created from params", function() {
+        var mixRuleSet = new OncoprintRuleSet(mixParams);
+        assert.equal(mixRuleSet.constructor.name, "GradientCategoricalRuleSet");
+    });
+
+    it("Formats gradient value", function() {
+        var mixRuleSet = new OncoprintRuleSet(mixParams);
+        var elements = mixRuleSet.apply([gradientDatumLargest, gradientDatumSmallest, naDatum], 12, 12);
+        assert.equal(elements.length, 3);
+        assert.equal(elements[0][0].fill,'rgba(0,255,0)');
+        assert.equal(elements[1][0].fill,'rgba(255,0,0,NaN)');
```

Comment:
why would `NaN` be a desired output?

Reply:
Well, this is the value returned by the existing oncoprintjs code that I did not touch. Having a NaN value for opacity is something I can live with, and I did not bother to inspect existing code. Is having NaN in there a problem?

Reply:
yes, that will probably break something, e.g. svg. I don't know why there should be a NaN there

Reply:
also, line 48 says `rgba` but it only has 3 elements

Reply:
okay the problem is params.colors needs to be 4-element arrays

Reply:
Ok, I updated the test as per your suggestion.

Reply:
I updated the test as per your suggestion.